motionplan/constraint.go

```diff
@@ -146,11 +146,7 @@ func NewCollisionConstraint(
 	if err != nil {
 		return nil
 	}
-	spaceEntities, err := NewSpaceCollisionEntities(interactionSpaces)
```

Comment:
This is an opportunity to make our collision system massively more readable.

We can take now as an opportunity to rip out `SpaceCollisionEntities` entirely.

Reply:
Completely agreed. That tripped me up a bunch when looking through this initially. Should just be `CollisionEntities` for all incarnations of the concept.